nncf/experimental/onnx/datasets/imagenet_dataset.py

```diff
@@ -35,25 +43,63 @@ def __len__(self):
         return len(self.dataset)
 
 
-def create_imagenet_torch_dataset(dataset_dir: str,
-                                  input_shape: List[int],
-                                  mean=(0.485, 0.456, 0.406),
-                                  std=(0.229, 0.224, 0.225),
-                                  crop_ratio=0.875,
-                                  batch_size: int = 1,
-                                  shuffle: bool = True):
-    import torchvision
-    from torchvision import transforms
-    image_size = [input_shape[-2], input_shape[-1]]
+def infer_input_shape(model: ModelProto) -> Tuple[Optional[int], int, int, int]:
```

Comment:
Requiring a model as an input element when doing a `create_dataset` is a bit difficult to understand.
Instead, we can place extract `input_shape` from loaded model in `onnx_ptq_classification.py` before making dataset when `input_shape` is None.
What do you think?

Reply:
Remove `model` parameter from `create_dataset` and move infer_input_shape to the upper level `onnx_ptq_classification.py::run()`.